internal/juju/offers.go

```diff
@@ -246,12 +247,46 @@ func (c offersClient) DestroyOffer(input *DestroyOfferInput) error {
 	return nil
 }
 
-func findApplicationOffers(client *applicationoffers.Client, filter crossmodel.ApplicationOfferFilter) (*crossmodel.ApplicationOfferDetails, error) {
+// filterByEndpoints is filtering offers that match exactly the endpoints' names provided.
+// If no endpoints are provided, all offers are returned to match the API behaviour.
+// The reason why we rely on this custom filtering and not the API filtering is that
+// the API filtering doesn't work as expected when multiple endpoints filters are provided.
+// An endpoint filter is composed of three fields: Name, Interface and Role.
+// If we try to filter by two endpoints' names, the API will return no offers, because the internal
+// logic is doing an AND on the different fields of the endpoints filter. Specifying two fields with the same
+// field (ex. Name) will always result in no matches.
+func filterByEndpoints(offers []*crossmodel.ApplicationOfferDetails, endpoints []string) []*crossmodel.ApplicationOfferDetails {
+	if len(endpoints) == 0 {
+		return offers
+	}
+	slices.Sort(endpoints)
+
+	filtered := []*crossmodel.ApplicationOfferDetails{}
+	for _, offer := range offers {
+		if len(offer.Endpoints) != len(endpoints) {
+			continue
+		}
+		endpointsNames := make([]string, 0, len(offer.Endpoints))
+		for _, endpoint := range offer.Endpoints {
+			endpointsNames = append(endpointsNames, endpoint.Name)
+		}
+		slices.Sort(endpointsNames)
+
+		if slices.Equal(endpointsNames, endpoints) {
```

Comment:
i wonder if this is the right thing to do.. 

what happens if the offer has a superset of endpoints you require.. 
example: offer has the following endpoints `[e1, e2, e3]` and you try to filter for `[e1, e3]`

Reply:
ok, we're looking for an exact match.. let's just adjust the wording so it's not too misleading: we're not filtering, we're looking for an exact match..